docs/api/grpc/overview.md

```diff
@@ -1,16 +1,162 @@
 
-- [Data](data-node/api/v2/trading_data.proto.mdx): Query the full set of historical data generated by the network (subject to retention policies). Trades, orders, governance, deposits, withdrawals, delegations, and more are all stored by the data node in a PostgreSQL database and queried through this API.
+The protobuf definitions for the data node API can be found [here](https://github.com/vegaprotocol/vega/blob/develop/protos/sources/data-node/api/v2/trading_data.proto). They allow for read-only queries for trading data and historic states of the network. 
+
+### Core node API
+Core nodes are responsible for ensuring the consensus rules are met and that a consistent view of the network is seen. They present endpoints that give access to the basic state of the network (block time, block height etc), allow transactions to be submitted to the network and to subscribe to event streams so that changes of internal state can be seen.
+
+The protobuf definitions for the core node API can be found [here](https://github.com/vegaprotocol/vega/blob/develop/protos/sources/vega/api/v1/core.proto). They allow for interacting with the network and sending in transactions.
+
+As a data node acts as a proxy to its core node, all `CoreService` API such a sending in transactions are also available from the data node's RPC address.
+
+## Client stub examples
+
+As an example, the below is an RPC endpoint and messages for querying an order defined with protocol-buffers:
+```proto
+// Get order
+//
+// Get an order by its ID. An order's ID will be the SHA3-256 hash of the signature that the order was submitted with
+rpc GetOrder(GetOrderRequest) returns (GetOrderResponse) {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Orders"};
+}
+
+// Request that is sent when executing the query for getting a single order
+message GetOrderRequest {
+  // Order ID to retrieve order information for.
+  string order_id = 1 [(google.api.field_behavior) = REQUIRED];
+  // Historic version number of the order to return. If not set, the most current version will be returned.
+  optional int32 version = 2;
+}
+
+// Response received from the query for getting a single order
+message GetOrderResponse {
+  // Order details, if one was found.
+  vega.Order order = 1;
+}
+```
+
+The above protobuf definitions can be used to generate client and server code in a given language. This can be done by using the protocol-compiler `protoc` directly and supplier a output language, or by using CLI tools from the buf-ecosystem, namely `buf generate`. As an example, YAML configuration for compiling Vega's protobuf definition into Golang using `buf genreate` can be found here [here](https://github.com/vegaprotocol/vega/blob/develop/buf.gen.yaml)
+
+
+Once generated they can imported and used from the generated language specific client stubs as follows:
+
+<Tabs>
+<TabItem value="py" label="Python">
```

Comment:
I've done python and golang examples. I don't know if we know what other languages might be useful given our knowledge of people who have already integrated?

Reply:
@wwestgarth Seems like this will do for now, and we can add more as we get requests/more data